src/apis/category/category.controller.js

```diff
@@ -0,0 +1,168 @@
+import { HTTP } from "../../constants/http-code.js";
+import { db } from "../../utils/db.server.js";
+
+class CategoryController {
+  getAll = async (req, res, next) => {
+    try {
+      const categories = await db.category.findMany({
+        include: {
+          foods: {
+            select: {
+              id: true,
+            },
+          },
+        },
+      });
+
+      const responseData = categories.map((category) => {
+        return {
+          id: category.id,
+          name: category.name,
+          imageId: category.imageId,
+          quantity: category.foods.length,
+        };
+      });
+
+      res.status(200).json({
+        categories: responseData,
+      });
+    } catch (error) {
+      console.log(error);
+      next(error);
+    }
+  };
+
+  add = async (req, res, next) => {
+    const { name, imageId } = req.body;
+    if (!name) {
+      res.status(HTTP.BAD_REQUEST).json({
+        message: "Invalid body request",
+      }); 
+    }
+
+    try {
+      const newCategory = await db.category.create({
+        data: {
+          name,
+          imageId
+        },
+      });
+
+      res.status(200).json({
+        message: "Category added successfully",
+        data: newCategory,
+      });
+    } catch (error) {
+      next(error)
+    }
+  };
+
+  remove = async (req, res, next) => {
+    const { id } = req.params;
+    if (!id) res.sendStatus(HTTP.BAD_REQUEST);
+
+    try {
+      const category = await db.category.findUnique({
+        where: {
+          id: parseInt(id),
+        },
+      });
+      if (!category)
+        res.status(404).json({
```

Comment:
It should be `return res.status(404)` because the code following it will continue to be executed